simplestore/src/main/java/com/uber/simplestore/impl/SimpleStoreImpl.java

```diff
@@ -191,6 +194,62 @@
         orderedIoExecutor);
   }
 
+  @Override
+  public ListenableFuture<Object> deleteAllNow() {
+    SimpleStoreFactory.flushAndClearRecursive(this);
+
+    return Futures.submitAsync(
+        () -> {
+          if (scopedDirectory != null) {
+            recursiveDelete(scopedDirectory);
+          }
+          return Futures.immediateFuture(null);
+        },
+        orderedIoExecutor);
+  }
+
+  /** Only call from the orderedIoExecutor. */
+  void clearCache() {
+    cache.clear();
+  }
+
+  /**
+   * Cause all items in the queue to fail out, then run something before enabling the queue again
+   */
+  void flushQueueThenRun(Exception exception, Runnable runnable) {
+    if (flush != null) {
+      throw new IllegalStateException();
+    }
+    this.flush = exception;
+    orderedIoExecutor.execute(
+        () -> {
+          runnable.run();
+          this.flush = null;
+        });
+  }
+
+  void moveAway() {
+    if (scopedDirectory == null) {
+      return;
+    }
+    File newLocation = new File(scopedDirectory.getAbsolutePath() + ".bak");
+    if (!scopedDirectory.renameTo(newLocation)) {
+      Log.e(getClass().getName(), "moveAway rename failed");
```

Comment:
I don't think you want to update the scopedDirectory in this case. add return statement?